ios/Classes/codec/AblyFlutterReader.m

```diff
@@ -145,6 +145,9 @@ use an explicitly received null from Dart (manifesting this side as a nil id)
     // track @ https://github.com/ably/ably-flutter/issues/14
 
     [clientOptions addAgent:@"ably-flutter" version:FLUTTER_PACKAGE_PLUGIN_VERSION];
+    ON_VALUE(^(const id value) {
```

Comment:
It's not actually a nil-check here. In ObjC there are 3 macros used to read values from JSON (there's a separate one for bool, due to ObjC treating bool values as numbers 😄 ):

```objc
#define ON_VALUE(BLOCK, DICTIONARY, JSON_KEY) { \
const id value = [DICTIONARY objectForKey: JSON_KEY]; \
if (value) { \
BLOCK(value); \
} \
}

#define READ_VALUE(OBJECT, PROPERTY, DICTIONARY, JSON_KEY) { \
ON_VALUE(^(const id value) { OBJECT.PROPERTY = value; }, DICTIONARY, JSON_KEY); \
}

#define READ_BOOL(OBJECT, PROPERTY, DICTIONARY, JSON_KEY) { \
ON_VALUE(^(const id number) { OBJECT.PROPERTY = [number boolValue]; }, DICTIONARY, JSON_KEY); \
}
```

You can see that internally they all call `ON_VALUE` anyway, the only difference is that `READ_VALUE` writes the value to object property. In this case, we do not want that behavior, because we need to call `ClientOptions.addAgent` method instead of just setting the property. I agree this may not be the safest option to encode and decode values, but this is just how the serialization currently works